tomopy/io/exchange.py

```diff
@@ -117,17 +131,79 @@ def read_als_832(fname, ind_tomo=None):
             nproj = int(re.findall(r'\d+', line)[0])
         if '-num_bright_field' in line:
             nflat = int(re.findall(r'\d+', line)[0])
+        if '-i0cycle' in line:
+            inter_bright = int(re.findall(r'\d+', line)[1])
         if '-num_dark_fields' in line:
             ndark = int(re.findall(r'\d+', line)[0])
     contents.close()
-
     if ind_tomo is None:
         ind_tomo = range(0, nproj)
-    ind_flat = range(0, nflat)
-    ind_dark = range(0, ndark)
+    if doNorm:
+        ind_flat = range(0, nflat)
+        if inter_bright > 0:
+            ind_flat = range(0, nproj, inter_bright)
+            flat_name = fname + 'bak_0000_0000.tif'
+        ind_dark = range(0, ndark)
+    # Read image data from tiff stack.
     tomo = tio.read_tiff_stack(tomo_name, ind=ind_tomo, digit=4)
-    flat = tio.read_tiff_stack(flat_name, ind=ind_flat, digit=4)
-    dark = tio.read_tiff_stack(dark_name, ind=ind_dark, digit=4)
+    if doNorm:
+    # Adheres to 8.3.2 flat/dark naming conventions: 
+    # ----Flats----
+    # root_namebak_xxxx_yyyy
+    # For datasets that take flat at the start and end of its scan,
+    # xxxx is in incrementals of one, and yyyy is either 0000 or the last projection.
+    # For datasets that take flat while they scan (when the beam fluctuates during scans),
+    # xxxx is always 0000, and yyyy is in intervals given by log file.
+ 
+        if inter_bright == 0:
+            list_flat = tio._list_file_stack(flat_name, ind_flat, digit=4)
+            for x in ind_flat:
+	        body = os.path.splitext(list_flat[x])[0]
+	        ext = os.path.splitext(list_flat[x])[1]
+                body += '_0000'
```

Comment:
Apparently they didn't like the way I commented, will fix it tomorrow.



On Wed, Jun 10, 2015 at 7:10 PM, Doga Gursoy <notifications@github.com>
wrote:

> In tomopy/io/exchange.py
> <https://github.com/dgursoy/tomopy/pull/1#discussion_r32184954>:
>
> > -    dark = tio.read_tiff_stack(dark_name, ind=ind_dark, digit=4)
> > +    if doNorm:
> > +    # Adheres to 8.3.2 flat/dark naming conventions:
> > +    # ----Flats----
> > +    # root_namebak_xxxx_yyyy
> > +    # For datasets that take flat at the start and end of its scan,
> > +    # xxxx is in incrementals of one, and yyyy is either 0000 or the last projection.
> > +    # For datasets that take flat while they scan (when the beam fluctuates during scans),
> > +    # xxxx is always 0000, and yyyy is in intervals given by log file.
> > +
> > +        if inter_bright == 0:
> > +            list_flat = tio._list_file_stack(flat_name, ind_flat, digit=4)
> > +            for x in ind_flat:
> > +	        body = os.path.splitext(list_flat[x])[0]
> > +	        ext = os.path.splitext(list_flat[x])[1]
> > +                body += '_0000'
>
> Indentation seems to contain mixed spaces and tabs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dgursoy/tomopy/pull/1/files#r32184954>.
>